src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -129,6 +132,14 @@ def eq(self, y):
 
         """
 
+        if not (torch.cat(self.shares) == torch.cat(y.shares)).all():
```

Comment:
Q: Why it is needed to call `torch.cat`? and not simply do `if not (self.shares == y.shares)`

Reply:
It was throwing an error `E       RuntimeError: Boolean value of Tensor with more than one value is ambiguous` so I cast each of them into a torch tensor and performed the operation. self.shares and y.shares are a list of tensors, now they become 2d torch tensors after using cat.